Shared/Extensions/UI Extensions/UICollectionView+Extensions.swift

```diff
@@ -10,6 +10,38 @@ import Foundation
 
 extension UICollectionView {
 
+    /// Returns the indexpath of the item whose cell is closest to the center of this collectionview. If there are no items visible, then nil is returned.
+    func getCentermostVisibleIndex() -> IndexPath? {
```

Comment:
Used to determine which message we're sending to. Get center x doesn't work because the cell sizes are the width of the collectionview.